spack/packages/dla-future/package.py

```diff
@@ -99,6 +99,10 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("pika +cuda", when="+cuda")
     depends_on("pika +rocm", when="+rocm")
 
+    for cxxstd in ("20", "23"):
+        conflicts(f"^pika cxxstd={cxxstd}", when="@:0.6 +cuda ^pika@:0.29")
```

Comment:
I just realized that we have another conflict that makes `dla-future@0.6 +cuda ^pika@0.30` anyway impossible: https://github.com/eth-cscs/DLA-Future/blob/0bbec23b8e6ec793a554cd0d035a57ebef0bd94a/spack/packages/dla-future/package.py#L95. Do you think it's worth keeping the pika version constraint here anyway despite the overlap?

Reply:
IMO constrains serves as documentation as well. I would keep it.